pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -3773,9 +3773,9 @@ private void validatePartitionTopicUpdate(String topicName, int numberOfPartitio
         TopicName partitionTopicName = TopicName.get(domain(), namespaceName, topicName);
         PartitionedTopicMetadata metadata = getPartitionedTopicMetadata(partitionTopicName, false, false);
         int oldPartition = metadata.partitions;
-        String prefix = topicName + TopicName.PARTITIONED_TOPIC_SUFFIX;
+        String prefix = partitionTopicName.getPartitionedTopicName() + TopicName.PARTITIONED_TOPIC_SUFFIX;
         for (String exsitingTopicName : existingTopicList) {
-            if (exsitingTopicName.contains(prefix)) {
+            if (exsitingTopicName.startsWith(prefix)) {
```

Comment:
This change won't work properly. The topic name will be in format `persistent://tenant/cluster/namespace/topicName` .

Reply:
There is a PR #11148 in progress in this same area. I'd suggest contributing to #11148 to fix this issue #11685. One form of contributing is making sure that this problem gets fixed with the refactored implementation for checking topic existence.

Reply:
> This change won't work properly. The topic name will be in format `persistent://tenant/cluster/namespace/topicName` .

yes I have considered that.  in this PR，I changed to the prefix = topicName + TopicName.PARTITIONED_TOPIC_SUFFIX, it's the full topicName contain 'persistent://tenant/cluster/namespace/topicName';

Reply:
> There is a PR #11148 in progress in this same area. I'd suggest contributing to #11148 to fix this issue #11685. One form of contributing is making sure that this problem gets fixed with the refactored implementation for checking topic existence.

it's full name `String prefix = partitionTopicName.getPartitionedTopicName() + TopicName.PARTITIONED_TOPIC_SUFFIX`

Reply:
> This change won't work properly. The topic name will be in format `persistent://tenant/cluster/namespace/topicName` .

changed to the prefix = partitionTopicName.getPartitionedTopicName() + TopicName.PARTITIONED_TOPIC_SUFFIX , it's the full topicName contain 'persistent://tenant/cluster/namespace/topicName';

Reply:
> same are

I don't think it's the same area, `validatePartitionTopicUpdate` only used when updating partition of topic. it's not the same logic.

Reply:
@lhotari I don't think this is the same issue as #11148 wants to fix. This is a bug here, #11148 is trying to improve the topic creation logic.